main/src/cgeo/geocaching/ImageSelectActivity.java

```diff
@@ -196,129 +196,38 @@ private void syncEditTexts() {
     }
 
     private void selectImageFromCamera() {
-        final File file = ImageUtils.getOutputImageFile();
-        if (file == null) {
-            showFailure();
-            return;
-        }
-
-        image = new Image.Builder().setUrl(Uri.fromFile(file)).build();
-        if (image.isEmpty()) {
-            showFailure();
-            return;
-        }
 
-        // create Intent to take a picture and return control to the calling application
-        final Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-        final Context context = getApplicationContext();
-        final Uri uri = FileProvider.getUriForFile(
-                context,
-                context.getString(R.string.file_provider_authority),
-                file);
-        intent.putExtra(MediaStore.EXTRA_OUTPUT, uri); // set the image file name
-
-        // start the image capture Intent
-        startActivityForResult(intent, SELECT_NEW_IMAGE);
+        imageActivityHelper.getImageFromCamera(-1, false, false, img -> {
+            deleteImageFromDeviceIfNotOriginal(image);
+            image = img;
+            loadImagePreview();
+        });
     }
 
     private void selectImageFromStorage() {
-        final Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
-        intent.setType("image/*");
 
-        startActivityForResult(Intent.createChooser(intent, "Select Image"), SELECT_STORED_IMAGE);
+        //TODO: revert to single
```

Comment:
something yet to be done, or an outdated comment?

Reply:
Danm, I forgot to remove the TODO tag. But the change itself was done. I can remove the false comment tomorrow (don't have access to source right now)

Reply:
> something yet to be done, or an outdated comment?

Todo is now removed

Reply:
thanks